src/inc/msquic.h

```diff
@@ -276,6 +276,52 @@ typedef struct QUIC_NEW_CONNECTION_INFO {
     const char* ServerName;
 } QUIC_NEW_CONNECTION_INFO;
 
+#define QUIC_CIPHER_SUITE_TLS_AES_128_GCM_SHA256          0x1301
+#define QUIC_CIPHER_SUITE_TLS_AES_256_GCM_SHA384          0x1302
+
+#ifdef CALG_AES_128
+#define QUIC_ALG_AES_128       CALG_AES_128
+#else
+#define QUIC_ALG_AES_128       0x660E
+#endif
+#ifdef CALG_AES_256
+#define QUIC_ALG_AES_256       CALG_AES_256
+#else
+#define QUIC_ALG_AES_256       0x6610
+#endif
+
+#ifdef CALG_SHA_256
+#define QUIC_ALG_SHA_256       CALG_SHA_256
+#else
+#define QUIC_ALG_SHA_256       0x800C
+#endif
+#ifdef CALG_SHA_384
+#define QUIC_ALG_SHA_384       CALG_SHA_384
+#else
+#define QUIC_ALG_SHA_384       0x800D
+#endif
+
+#define QUIC_TLS_UNKNOWN        0
+#ifdef SP_PROT_TLS1_3
+#define QUIC_TLS1_3             SP_PROT_TLS1_3
+#else
+#define QUIC_TLS1_3             0x3000
+#endif
+
+//
+// All the available information describing a handshake.
+//
+typedef struct QUIC_HANDSHAKE_INFO {
+    int32_t TlsProtocolVersion;
+    int32_t CipherAlgorithm;
+    int32_t CipherStrength;
+    int32_t Hash;
+    int32_t HashStrength;
+    int32_t KeyExchangeAlgorithm;
+    int32_t KeyExchangeStrength;
+    int32_t CipherSuite;
```

Comment:
Personally, I don't like that these are just random `int32_t`. I'd much rather `enum`s for most of these. Just because Schannel does it this way doesn't make it right... Thoughts?

Reply:
Good idea. Done. It also implicitly solved the definition issue.